### ISSUE TITLE
Editor: Add a contrast check to the "style" panel

### ISSUE THREAD

**Issue by samwhale** [GoogleForCreators/web-stories-wp#7230]
## Feature Description

If a user were to pick a style that would make the text hard to read, announce `This color combination may be hard for people to read`.

The `useLiveRegion` hook can be used to announce the message to the user. However, we will need to implement some sort of contrast check in the panel.

## Alternatives Considered

n/a

## Additional Context

Split off from #6059 due to scope creep

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

**Comment by miina** [comment]
@BrittanyIRL This is currently assigned to Prometheus, however, it looks like it's a follow-up to an issue Pea Pod was working on, seems like it might make more sense for your pod to work on it (if that's the case). Thoughts?

**Comment by BrittanyIRL** [comment]
@miina - happy to take this!

**Comment by BrittanyIRL** [comment]
Should there be UX added here in the style panel to show when the contrast is bad or is it ok to just rely on the checklist which is already tracking it?

**Comment by swissspidy** [comment]
I think there should be a warning in the panel so you can fix it immediately

**Comment by BrittanyIRL** [comment]
We synced about this 11/30 with @agingoldseco who is going to work on a mock. It'll  probably just be some text and an icon below color picker when warning is triggered.

**Comment by agingoldseco** [comment]
@BrittanyIRL Here's the (screenshot of the) mockup for the text/icon that will pop-up with a bad color combo. I found this yellow icon in the design system for "low warning"—think it would make sense to use that? 

Also, what are your thoughts on having (perhaps) the whole box link to something about ADA compliance? Noticed that isn't in the checklist either.

<img width="637" alt="Screen Shot 2021-12-07 at 4 16 04 PM" src="https://user-images.githubusercontent.com/90645922/145107663-cb2911e3-f28a-46e7-8b96-f32ae8a9287b.png">

**Comment by BrittanyIRL** [comment]
Icon makes sense! 

Down to link - that should get run past @choumx and @swissspidy as they might have ideas about what to specifically link to.

**Comment by BrittanyIRL** [comment]
@choumx / @gajasinski what do you think about us pulling this into our extra sprint this release cycle?

**Comment by swissspidy** [comment]
ADA is very US-specific :-) This would need to be some more general documentation.

That said:

For comparison, WP uses `This color combination may be hard for people to read. Try using a brighter background color and/or a darker text color.`

Personally I don't think linking to some documentation that explains that in more complicated language adds any value. So I would not add any link.

**Comment by BrittanyIRL** [comment]
Thank you, Pascal ❤️

**Comment by dreamofabear** [comment]
+1 to Pascal's comment.

> @BrittanyIRL Here's the (screenshot of the) mockup for the text/icon that will pop-up with a bad color combo. I found this yellow icon in the design system for "low warning"—think it would make sense to use that?
> 
> Also, what are your thoughts on having (perhaps) the whole box link to something about ADA compliance? Noticed that isn't in the checklist either.
> 
> <img alt="Screen Shot 2021-12-07 at 4 16 04 PM" width="637" src="https://user-images.githubusercontent.com/90645922/145107663-cb2911e3-f28a-46e7-8b96-f32ae8a9287b.png">

@agingoldseco Quick feedback about this mock -- I like that it's contextual (near the affected UI), but it introduces a (good) new precedent for checklist feedback living outside of the checklist popup.

Do we think this specific pattern is extensible for other checklist items? If we're adding a new pattern, ideally it should be generalizable. /cc @aaskedall

**Comment by agingoldseco** [comment]
@choumx I agree there are other instances we can use it, but we'll have to be _super_ choosey about how we do—don't want users to feel like we're yelling at them. Especially for something they can't or really don't want to fix, or that came from a template. Maybe we can use it for something like truly egregiously low res images?

**Comment by felipebochehin87** [comment]
Tested against https://stories-qa-wordpress-amp.pantheonsite.io/wp-admin/, using PR #10074.

1. Added a text to the canvas
2. Changed text color to make it hard to read - warning message displayed on both "Accessibility" (on the checklist), below the color picker and on background too
3. Changed text color to make it more visible - warnings disappear
4. Repeated steps above with screen reader on - warnings are displayed and hidden (and also read by voice over) as expected

Background:

![background](https://user-images.githubusercontent.com/96055551/150187327-8f246541-0716-4756-adfd-6bd6a4341e07.png)

Color picker:

![color picker](https://user-images.githubusercontent.com/96055551/150187409-f8c89a65-7173-48a5-98fd-7ea94004cbbf.png)

Checklist:

![checklist](https://user-images.githubusercontent.com/96055551/150187502-04ad9b7d-4f02-4435-87fb-a96351a70072.png)
